From fbf5f9bce43e19f8827e5cdef0e456b74ccc2f7d Mon Sep 17 00:00:00 2001
Message-Id: <fbf5f9bce43e19f8827e5cdef0e456b74ccc2f7d@dist-git>
From: Pavel Hrdina <phrdina@redhat.com>
Date: Tue, 17 Jan 2023 10:08:08 +0100
Subject: [PATCH] domain_validate: drop cpu.shares cgroup check

This check is done when VM is defined but doesn't take into account what
cgroups version is currently used on the host system so it doesn't work
correctly.

To make proper check at this point we would have to figure out cgroups
version while defining a VM but that will still not guarantee that the
VM will start correctly in the future as the host may be rebooted with
different cgroups version.

Signed-off-by: Pavel Hrdina <phrdina@redhat.com>
Reviewed-by: Martin Kletzander <mkletzan@redhat.com>
(cherry picked from commit 38af6497610075e5fe386734b87186731d4c17ac)

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=2037998

Signed-off-by: Pavel Hrdina <phrdina@redhat.com>
---
 src/conf/domain_validate.c | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/src/conf/domain_validate.c b/src/conf/domain_validate.c
index 5a9bf20d3f..39d924d4ed 100644
--- a/src/conf/domain_validate.c
+++ b/src/conf/domain_validate.c
@@ -1725,16 +1725,6 @@ virDomainDefOSValidate(const virDomainDef *def,
 static int
 virDomainDefCputuneValidate(const virDomainDef *def)
 {
-    if (def->cputune.shares > 0 &&
-        (def->cputune.shares < VIR_CGROUP_CPU_SHARES_MIN ||
-         def->cputune.shares > VIR_CGROUP_CPU_SHARES_MAX)) {
-        virReportError(VIR_ERR_CONFIG_UNSUPPORTED,
-                       _("Value of cputune 'shares' must be in range [%llu, %llu]"),
-                         VIR_CGROUP_CPU_SHARES_MIN,
-                         VIR_CGROUP_CPU_SHARES_MAX);
-        return -1;
-    }
-
     CPUTUNE_VALIDATE_PERIOD(period);
     CPUTUNE_VALIDATE_PERIOD(global_period);
     CPUTUNE_VALIDATE_PERIOD(emulator_period);
-- 
2.39.1

